modules/notify/main/index.coffee

```diff
@@ -1,33 +1,32 @@
 # utility method for setting up notifications
 setupNotification = (notification, selection) ->
-  if notification.options.icon? and notification.options.icon.length > 0
-    selection.append('div')
-      .class('hx-notification-icon-container')
-      .append('i')
-        .class('hx-notification-icon ' + notification.options.icon)
+  icon = if notification.options.icon? and notification.options.icon.length > 0
+    hx.detached('div').class('hx-notification-icon-container')
+      .add(hx.detached('i').class('hx-notification-icon ' + notification.options.icon))
 
-  selection
-    .append('div')
-      .class('hx-notification-text')
-      .text(notification.message)
+  content = hx.detached('div').class('hx-notification-content')
+  notification.options.renderer(content.node(), notification.message)
 
   if notification.options.pinnable
-    notification.domPin = selection
-      .append('div')
-        .class('hx-notification-icon-container hx-notification-pin')
-        .on 'click', 'hx.notify', -> togglePin(notification)
-
-    notification.domPin.append('i')
-      .attr('class', 'hx-icon hx-icon-thumb-tack')
+    pin = hx.detached('div')
+      .class('hx-notification-icon-container hx-notification-pin')
+      .on('click', 'hx.notify', -> togglePin(notification))
+      .add(hx.detached('i').attr('class', 'hx-icon hx-icon-thumb-tack'))
```

Comment:
I was going to use the fluid api for this but I'm not sure it's worth doing in this case